tao_compiler/mlir/disc/transforms/disc_algebra_simplifier.cc

```diff
@@ -345,6 +345,31 @@ struct BroadCastInDimOfReshapeOpCanonicalizationPattern
   }
 };
 
+template <typename OpTy>
+struct SimplifierFromElementsPattern : public OpRewritePattern<OpTy> {
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter& rewriter) const override {
+    auto loc = op->getLoc();
+    Value input = op->getOperand(0);
+    Value result = op->getResult(0);
+    auto extractOp = input.getDefiningOp();
+    if (!isa<tensor::ExtractOp>(extractOp)) {
+      return failure();
+    }
+    auto rankTy = op.getType().template dyn_cast<RankedTensorType>();
+    auto extractInput = extractOp->getOperand(0);
```

Comment:
What if  the source tensor of`tensor.extract` having `rank > 0`. For example, `tensor.extract %0[%c0, ...]: tensor<?x?xf32>`?

If you not want to process the other case, please check and early return.

Reply:
this pass only deal with input tensor of `tensor.extract` with `rank == 0`, I added some limitation and return early.